src/frontend/test_runner/tests/testdata/index_selection.yaml

```diff
@@ -119,3 +119,59 @@
   batch_plan: |
     BatchExchange { order: [], dist: Single }
       BatchScan { table: idx2, columns: [idx2.a, idx2.b, idx2.c], scan_ranges: [idx2.b = Decimal(Normalized(2))], distribution: UpstreamHashShard(idx2.b, idx2.a) }
+- sql: |
+    create table t1 (a int, b numeric, c bigint);
+    create index idx1 on t1(a);
+    create index idx2 on t1(b, a);
+    create index idx3 on t1(c);
+    select * from t1 where c = 1 and a < 10;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchLookupJoin { type: Inner, predicate: idx3.t1._row_id = t1._row_id AND (t1.a < 10:Int32), output: [t1.a, t1.b, t1.c] }
+        BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c = Int64(1)], distribution: SomeShard }
+  batch_local_plan: |
+    BatchLookupJoin { type: Inner, predicate: idx3.t1._row_id = t1._row_id AND (t1.a < 10:Int32), output: [t1.a, t1.b, t1.c] }
+      BatchExchange { order: [], dist: Single }
+        BatchScan { table: idx3, columns: [idx3.t1._row_id], scan_ranges: [idx3.c = Int64(1)], distribution: SomeShard }
+- sql: |
+    create table t1 (a int, b numeric, c bigint);
+    create index idx1 on t1(a);
+    create index idx2 on t1(b, a);
+    create index idx3 on t1(c);
+    select * from t1 where a = 1;
+  batch_plan: |
+    BatchExchange { order: [], dist: Single }
+      BatchLookupJoin { type: Inner, predicate: idx1.t1._row_id = t1._row_id, output: [t1.a, t1.b, t1.c] }
+        BatchScan { table: idx1, columns: [idx1.t1._row_id], scan_ranges: [idx1.a = Int32(1)], distribution: SomeShard }
+  batch_local_plan: |
+    BatchLookupJoin { type: Inner, predicate: idx1.t1._row_id = t1._row_id, output: [t1.a, t1.b, t1.c] }
+      BatchExchange { order: [], dist: Single }
+        BatchScan { table: idx1, columns: [idx1.t1._row_id], scan_ranges: [idx1.a = Int32(1)], distribution: SomeShard }
+- sql: |
+    create table t1 (a int, b numeric, c bigint);
+    create index idx1 on t1(a);
+    create index idx2 on t1(b, a);
+    create index idx3 on t1(c);
+    select * from t1 where a = 1 and b = 2;
```

Comment:
Another question: Can our lookup join now choose 2 or more index underneath at the same time? I'm not sure whether other Database Otimizer can do this.

Reply:
Yes, I think what you mean is merge index. We can support it later. I already leave a `TODO` within the `IndexSelectionRule`.

## Example:
```sql
create table t1 (a int, b numeric, c bigint);
create index idx1 on t1(a);
create index idx3 on t1(c);
select * from t1 where a = 1 or c = 1; // merge index can use idx1 and idx3 at the same time.
```